#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Fix alpha build failure

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- sysdeps/alpha/divrem.h	6 Jul 2001 04:55:45 -0000	1.8
+++ sysdeps/alpha/divrem.h	15 Jun 2002 20:53:37 -0000	1.9
@@ -1,4 +1,4 @@
-/* Copyright (C) 1996, 1997 Free Software Foundation, Inc.
+/* Copyright (C) 1996,97,2002 Free Software Foundation, Inc.
    Contributed by David Mosberger (davidm@cs.arizona.edu).
    This file is part of the GNU C Library.
 
@@ -86,6 +86,7 @@
 
 	.align 3
 UFUNC_NAME:
+$udiv_entry:
 	lda	sp, -STACK(sp)
 	.frame	sp, STACK, retaddr, 0
 #ifdef PROF
@@ -206,7 +207,7 @@
 	cmovge	AT, AT, arg2
 
 	/* Do the unsigned division.  */
-	bsr	retaddr, UFUNC_NAME
+	bsr	retaddr, $udiv_entry
 
 	/* Restore originals and adjust the sign of the result.  */
 	ldq	arg1, 0(sp)
